Source/WebGPU/WebGPU/Queue.mm

```diff
@@ -121,9 +121,30 @@
     return buffer;
 }
 
+void Queue::makeInvalid()
+{
+    m_commandQueue = nil;
+    for (auto& [_, callbackVector] : m_onSubmittedWorkScheduledCallbacks) {
+        for (auto& callback : callbackVector)
+            callback();
+    }
+    for (auto& [_, callbackVector] : m_onSubmittedWorkDoneCallbacks) {
+        for (auto& callback : callbackVector)
+            callback(WGPUQueueWorkDoneStatus_Success);
+    }
+
+    m_onSubmittedWorkScheduledCallbacks.clear();
+    m_onSubmittedWorkDoneCallbacks.clear();
+}
+
 void Queue::onSubmittedWorkDone(CompletionHandler<void(WGPUQueueWorkDoneStatus)>&& callback)
 {
     // https://gpuweb.github.io/gpuweb/#dom-gpuqueue-onsubmittedworkdone
+    auto devicePtr = m_device.get();
+    if (!devicePtr || !devicePtr->isValid() || devicePtr->isLost()) {
+        callback(WGPUQueueWorkDoneStatus_Success);
```

Comment:
Is this really a success?

Reply:
Good question, I think it should be WGPUQueueWorkDoneStatus_DeviceLost instead